multiqc/modules/biscuit/biscuit.py

```diff
@@ -0,0 +1,688 @@
+            for l in f.splitlines():
+                fields = l.strip().split('\t')
+                if fields[2] not in ['C','R'] or fields[0] not in ['1','2']:
+                    return {}
+                if fields[0] == '1':
+                    r1[fields[2]][int(fields[1])] = int(fields[3])
+                if fields[0] == '2':
+                    r2[fields[2]][int(fields[1])] = int(fields[3])
+            r1rate = OrderedDict()
+            for k in sorted(r1['C'].keys()):
+                if k in r1['R']:
+                    r1rate[k] = float(r1['R'][k])/(r1['R'][k]+r1['C'][k])*100
+            r2rate = {}
+            for k in sorted(r2['C'].keys()):
+                if k in r2['R']:
+                    r2rate[k] = float(r2['R'][k])/(r2['R'][k]+r2['C'][k])*100
+
+            data = {'1':r1rate, '2':r2rate}
+
+        elif fn.endswith('_beta_table.txt'):
+            sumcounts = 0
+            for l in f.splitlines():
+                fields = l.split('\t')
+                sumcounts += int(fields[1])
+            
+            data = {}
+            for l in f.splitlines():
+                fields = l.split('\t')
+                data[int(fields[0])] = int(fields[1]) / float(sumcounts)
+            
+        return data
+
+    def chart_biscuit_retention(self):
+
+        mdata1 = dict([(k.replace('_totalReadConversionRate.txt',''),v) for k, v in self.mdata['retention'].items() if k.endswith('_totalReadConversionRate.txt')])
+        mdata2 = dict([(k.replace('_totalBaseConversionRate.txt',''),v) for k, v in self.mdata['retention'].items() if k.endswith('_totalBaseConversionRate.txt')])
+        pdata = dict([(sid, dict(list(dd.items()) + list(mdata2[sid].items()))) for sid, dd in mdata1.items()])
+
+        pheader = OrderedDict()
+        pheader['ca'] = {'title':'r.CpA', 'format':'{:,.2g}', 'min':0, 'max':100, 'description':'CpA', 'suffix':'%'}
+        pheader['cc'] = {'title':'r.CpC', 'format':'{:,.2g}', 'min':0, 'max':100, 'description':'CpC', 'suffix':'%'}
+        pheader['cg'] = {'title':'r.CpG', 'format':'{:,.2g}', 'min':0, 'max':100, 'description':'CpG', 'suffix':'%'}
+        pheader['ct'] = {'title':'r.CpT', 'format':'{:,.2g}', 'min':0, 'max':100, 'description':'CpT', 'suffix':'%'}
+        pheader['bca'] = {'title':'b.CpA', 'format':'{:,.2g}', 'min':0, 'max':100, 'description':'CpA', 'suffix':'%'}
+        pheader['bcc'] = {'title':'b.CpC', 'format':'{:,.2g}', 'min':0, 'max':100, 'description':'CpC', 'suffix':'%'}
+        pheader['bcg'] = {'title':'b.CpG', 'format':'{:,.2g}', 'min':0, 'max':100, 'description':'CpG', 'suffix':'%'}
+        pheader['bct'] = {'title':'b.CpT', 'format':'{:,.2g}', 'min':0, 'max':100, 'description':'CpT', 'suffix':'%'}
+
+        self.add_section(
+            name = 'Cytosine Retention',
+            anchor = 'biscuit-retention',
+            description = "<p>This plot shows cytosine retention rate. 'r.' stands for read-averaging and 'b.' stands for 'base-averaging.</p>",
```

Comment:
You can use markdown backticks for `r` which will look nicer that 'r' I think. No HTML tags needed.

Reply:
It would be great to have help text explaining this too. I've worked with bisulfite data quite a bit and I'm not sure what cytosine retention means.

Reply:
Thanks for the suggestion. I switched to `r` and removed `<p>`. I added help text to explain on cytosine retention/conversion.